File: java-streams/examples/java/src/main/java/streams/Parallel.java
package streams;

import java.util.LinkedList;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;

public class Parallel {

    static String allNames(Stream<Person> persons) {

        return persons
                .map(person -> {
                    try {
                        Thread.sleep(1);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                    return person.getFirstName().toUpperCase() + " " + person.getLastName().toUpperCase();
                })
                .collect(Collectors.joining());
    }

    public static void main(String[] args) {
        System.out.println("Prepare");
        List<Person> people = IntStream.range(0, 10000)
                .mapToObj(id -> new Person(id, "First " + id, "Last " + id, true, new LinkedList<>()))
                .collect(Collectors.toList());

        System.out.println("GO serial");
        long start = System.currentTimeMillis();
        String result = allNames(people.stream()); //.parallel()
        System.out.println("time = " + ((System.currentTimeMillis() - start) / 1000.0));

        System.out.println("GO parallel");
        long start2 = System.currentTimeMillis();
        String result2 = allNames(people.stream().parallel());
        System.out.println("time = " + ((System.currentTimeMillis() - start2) / 1000.0));

    }
}

File: java-streams/examples/kotlin/src/main/kotlin/streams/Parallel.kt
package streams

import java.util.LinkedList
import java.util.stream.Collectors
import java.util.stream.IntStream
import java.util.stream.Stream

object Parallel {

    fun allNames(persons: Stream<Person>)=
            persons
                .map { (_, firstName, lastName) ->
                    try {
                        Thread.sleep(1)
                    } catch (e: InterruptedException) {
                        e.printStackTrace()
                    }

                    "${firstName.toUpperCase()} ${lastName.toUpperCase()}"
                }

}

fun main(args: Array<String>) {
    println("Prepare")
    val people = (0..10000)
            .map { id -> Person(id, "First " + id, "Last " + id, true, LinkedList()) }

    println("GO serial")
    val start = System.currentTimeMillis()
    val result = allNames(people.stream()) //.parallel()
    println("time = " + (System.currentTimeMillis() - start) / 1000.0)

    println("GO parallel")
    val start2 = System.currentTimeMillis()
    val result2 = allNames(people.parallelStream())
    println("time = " + (System.currentTimeMillis() - start2) / 1000.0)

}
